management-interface.go

```diff
@@ -39,21 +44,82 @@ func showDiskSpace(w http.ResponseWriter) error {
 		fmt.Fprintf(w, "Cannot show disk space at this time.\n")
 		return err
 	}
-	fmt.Fprintf(w, "%s\n", out)
+	fmt.Fprintf(w, "Disk space usage is: \n\n%s\n", out)
 	return nil
 
 }
 
+// Show the available network interfaces on this machine.
+func availableInterfaces(w http.ResponseWriter) error {
+
+	interfaces, err := net.Interfaces()
+
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(w, "Available network interfaces on this machine:\n\n")
+	for _, i := range interfaces {
+		if i.Name != "lo" {
+
+			fmt.Fprintf(w, "Testing %v...\n", i.Name)
+			fmt.Fprintf(w, "  Sending %v packets\n", NumPacketsToSend)
+
+			// Windows
+			//commandStr := "ping -n " + strconv.Itoa(NumPacketsToSend) + " -w 15 1.1.1.1"
+			//out, _ := exec.Command("cmd", "/C", commandStr).Output()
+			//pos := strings.Index(outStr, "Received =")
+			//if pos != -1 {
+			//	numPacketsReceivedStr := string(outStr[pos+11])
+
+			// Unix
+			commandStr := "ping -I " + i.Name + " -c " + strconv.Itoa(NumPacketsToSend) + " -n -W 15 1.1.1.1"
+			out, _ := exec.Command("sh", "-c", commandStr).Output()
+
+			// Did we get the output we'd expect if the network interface is up?
+			interfaceUp := false
+			outStr := string(out)
+			pos := strings.Index(outStr, "transmitted")
+			if pos != -1 {
+				numPacketsReceivedStr := string(outStr[pos+13])
+				fmt.Fprintf(w, "  Received %v packets\n", numPacketsReceivedStr)
+				numPacketsReceived, _ := strconv.Atoi(numPacketsReceivedStr)
+				if numPacketsReceived > 0 {
+					// I consider this interface to be "up".
+					interfaceUp = true
+				}
+			}
+			if interfaceUp {
+				fmt.Fprintf(w, "\n  %v is UP.\n\n", i.Name)
+			} else {
+				fmt.Fprintf(w, "\n  %v is DOWN.\n\n", i.Name)
+			}
+
+		}
+	}
+	return nil
+}
```

Comment:
Running ping tests over all the interfaces automatically isn't quite what I had in mind but it works for now. I was thinking you would list the interfaces along with their addresses and up status (see the Flags and FlagUp at https://golang.org/pkg/net/#Interface) and then have button next to each one to trigger a ping test.